### PR TITLE
Automate publishing to automation hub

### LINKED ISSUE #1579: Auto upload operator to operator hub upon new release
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that AWX Operator is open source software provided for free and that I might not receive a timely response.

### Feature Summary

A pipeline to auto upload the awx-operator to operator hub upon new releases to avoid forgetting to update the operator.

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
fixes https://github.com/ansible/awx-operator/issues/1579

##### SUMMARY
- Add github action to create PR to publish to automation hub
- Add AWX_AUTO_GITHUB_TOKEN to secret (not in PR)
- Add PREV_VERSION and print to Makefile

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by TheRealHaoLiu** [comment]
I accidently push the change to devel branch directly...

but HOW WAS I ABLE TO DO THAT @rooftopcellist

**Comment by TheRealHaoLiu** [comment]
TL;DR admin privilege with bypass branch protection permission...

disabling admin's permission to push to devel to prevent it for the future

image

**Review by rooftopcellist — APPROVED** [review]
